daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/RuntimeData.scala

```diff
@@ -763,12 +769,179 @@ final class ElementRuntimeData(
+    LocalDeclQName(None, local, NS(namespaceURI)), // namedQNameArg: => NamedQName,
+    false, // isRepresentedArg: => Boolean,
+    false, // couldHaveTextArg: => Boolean,
+    0, // alignmentValueInBitsArg: => Int,
+    false, // hasNoSkipRegionsArg: => Boolean,
+    null, // impliedRepresentationArg: => Representation,
+    null, // optIgnoreCaseArg: => Option[YesNo],
+    null, // optDefaultValueArg: => Option[AnyRef],
+    null, // optTruncateSpecifiedLengthStringArg: => Option[Boolean],
+    null, // outputValueCalcExprArg: => Option[CompiledExpression[AnyRef]],
+    Nope, // maybeBinaryFloatRepEvArg: => Maybe[BinaryFloatRepEv],
+    Nope, // maybeByteOrderEvArg: => Maybe[ByteOrderEv],
+    Nope, // maybeFillByteEvArg: => Maybe[FillByteEv],
+    Nope, // maybeCheckByteAndBitOrderEvArg: => Maybe[CheckByteAndBitOrderEv],
+    Nope, // maybeCheckBitOrderAndCharsetEvArg: => Maybe[CheckBitOrderAndCharsetEv],
+    false // isQuasiElementArg: => Boolean
+  ) {
+
+  override def toString() = Misc.getNameFromClass(this) + "(" + this.namedQName.toExtendedSyntax + ")"
+
+  @throws(classOf[java.io.IOException])
+  private def writeObject(out: java.io.ObjectOutputStream): Unit =
+    Assert.usageError("Not for serialization")
+}
+
+/**
+ * Used when unparsing to indicate that a next element event was detected that is
+ * unexpected.
+ */
+final class UnexpectedElementErrorERD(
+  optTRD: Option[TermRuntimeData],
+  local: String,
+  namespaceURI: String,
+  val allPossibleNQNs: Seq[QNameBase])
+  extends ErrorERD(local, namespaceURI) {
+
+  System.err.println("Creating.")
+  /**
+   * Causes unparse error with diagnostic about unexpected element.
+   *
+   * Pass argument true if the context is one where no element was expected
+   * (e.g., because an EndElement was expected.)
+   *
+   * Pass false if the Term's ordinary nextElementResolver list of possibilities
+   * is what was expected.
+   */
+  def toUnparseError(nothingWasExpected: Boolean = false) = {
+    val sqn = StepQName(None, name, thisElementsNamespace)
+    val sqnx = sqn.toExtendedSyntax
+    val allPossiblesString =
+      allPossibleNQNs.map { _.toExtendedSyntax }.mkString(", ")
+    val maybeLoc: Maybe[SchemaFileLocation] = Maybe.toMaybe(optTRD.map { _.schemaFileLocation })
+    allPossibleNQNs.length match {
+      case 0 | _ if (nothingWasExpected) =>
+        UnparseError(maybeLoc, Nope,
+          "Found next element %s, but no start of element is expected.",
+          sqnx)
+      case 1 =>
+        UnparseError(maybeLoc, Nope,
+          "Found next element %s, but expected %s.",
+          sqnx, allPossiblesString)
+      case _ =>
+        UnparseError(maybeLoc, Nope,
+          "Found next element %s, but expected one of %s.",
+          sqnx, allPossiblesString)
+    }
+  }
+}
+
+/**
+ * Used when unparsing to indicate that a next element event was detected that is
+ * unexpected.
+ */
+final class NamespaceAmbiguousElementErrorERD(
```

Comment:
No test insures this situation is detected, or the string "found mulitple matches" isn't in any test I can find. 

Need test to cover this case. 

Since there are no calls to toUnparseError however, this message can't get issued. 

So something is wrong because this isn't being invoked ever.

Reply:
Created  https://issues.apache.org/jira/browse/DAFFODIL-2236